### PR TITLE
RFR: Include request id in error objects

### PR DESCRIPTION
**Pull request by lakshmi-kannan**

### CODE REVIEW

File: lib/mixins/writable.js [hunk 1]
```diff
@@ -15,11 +15,16 @@ var Writable = {
       }, payload).then(function (res) {
 
         if (res.statusCode !== 201) {
-          throw {
+          var err = {
```
Comment (lakshmi-kannan) [inline lib/mixins/writable.js]: Should I call this ``errObj`` or ``err`` is fine?

**Comment by lakshmi-kannan** [comment]
```
[16:23:56] 'test-browser' errored after 14 s
[16:23:56] Error in plugin 'gulp-mocha-phantomjs'
test failed

npm ERR! Linux 3.16.0-30-generic
npm ERR! argv "/usr/bin/nodejs" "/usr/bin/npm" "install"
npm ERR! node v4.4.7
npm ERR! npm  v2.15.8
npm ERR! code ELIFECYCLE
npm ERR! st2client@0.4.5 prepublish: `gulp`
npm ERR! Exit status 1
```

happens on my box :/.

File: lib/mixins/writable.js (hunk 1 above)
Comment (enykeev) [inline lib/mixins/writable.js]: totally fine

File: lib/request.js [hunk 2]
```diff
@@ -58,11 +58,14 @@ var request = function (params, body) {
       return err;
     })
     .then(function (response) {
+      var contentType = response.headers['content-type'] || [];
+      var requestId = response.headers['X-Request-ID'] || null;
```
Comment (enykeev) [inline lib/request.js]: The reason phantomjs tests are failing is that in this particular environment, `response.headers` is undefined so you're basically getting a KeyError. The correct snippet should look somewhat like:

```
      var contentType = (response.headers || {})['content-type'] || [];
      var requestId = (response.headers || {})['X-Request-ID'] || null;
```

The other question however is why phantomjs didn't output the error to console and I'm going to fix that in the next PR.

**Comment by enykeev** [comment]
Now Circle would explain to you what went wrong in much greater detail.

**Comment by lakshmi-kannan** [comment]
@enykeev: Sweet. This should help me.

File: lib/request.js (hunk 2 above)
Comment (lakshmi-kannan) [inline lib/request.js]: The console messages are actually helpful. Thanks for fixing. Pushed a commit to fix the goof up.

**Pull request merged**